src/pkg/util/auth/token.go

```diff
@@ -28,10 +28,15 @@ const (
 // ReadToken reads a sylabs JWT auth token from a file
 func ReadToken(tokenPath string) (token, warning string) {
 	// check if token file exist
-	if _, err := os.Stat(tokenPath); os.IsNotExist(err) {
+	tokenFile, err := os.Stat(tokenPath)
+	if os.IsNotExist(err) {
 		return "", WarningTokenFileNotFound
 	}
 
+	if tokenFile.IsDir() {
```

Comment:
Don't test for dir - it could also be a socket or something else, and we don't want to add tests for everything it shouldn't be. Try to test instead that it is a readable normal file (or a link that can be resolved to a readable normal file)

Reply:
I think this check might be redundant, actually, since `ioutil.ReadFile()` will fail. The logic fix @ArangoGutierrez just applied in `build.go` should fix the issue there.  I'm not 100% sure about the Library side though.

For consistency, should the `"Couldn't read your Sylabs authentication token"` `string` below be a defined `const` like the other `warning` values?

Reply:
This `IsDir()` check should either be updated as @dctrud suggested, or removed completely (since `ioutil.ReadFile()` will catch the case where it isn't a file.)

Reply:
ioutil.ReadFile does not fail if given a dir... see https://play.golang.org/p/EHGX4lymOKQ